### DISCUSSION TITLE
Error: 'args.method' must be a non-empty string.

### DISCUSSION THREAD

**Discussion by VictorBlockchain** [WalletConnect/walletconnect-monorepo#3886] [Developer Support]
this worked just fine in V1
    let contract = await new web3.eth.Contract(mintABI.abi, mint);
    const result = await contract.methods.getDisplays().call();

now I keep getting the error.. Error: 'args.method' must be a non-empty string.

console.log of  contract.methods.getDisplays()

arguments
: 
[]
call
: 
(options, block) => {…}
createAccessList
: 
(options, block) => {…}
encodeABI
: 
() => {…}
estimateGas
: 
(options, returnFormat = web3_types__WEBPACK_IMPORTED_MODULE_4__.DEFAULT_RETURN_FORMAT) => {…}
send
: 
(options) => {…}
[[Prototype]]
: 
Object

console.log of methods
["getDisplays()"]

I'm using a nextjs application
  const { address, chainId, isConnected } = useWeb3ModalAccount()
  const { signer,walletProvider } = useWeb3ModalSigner()

wallet provider is passed to the service
  useEffect(() => {
    if (isConnected) {
        setWalletProvider(walletProvider);
        if(address){
            console.log('account set')
            setAccount(address);
            handleStartConnected(address)
        }
    }
  }, [walletProvider, isConnected]);
  
  provider is accessed from the service
  export function setWalletProvider(provider) {
  walletProvider = provider;
}
export function getWalletProvider() {
  if (!walletProvider) {
    throw new Error('Wallet provider is not set. Please call setWalletProvider(provider) before using getWalletProvider.');
  }
  
  return walletProvider;
}

const getWeb3 = () => {
  return new Promise(async(resolve, reject) => {
      let provider = getWalletProvider()
      web3 = new Web3(provider);
      // console.log(web3)
      resolve(web3)
  
  });
};

**Comment by glitch-txs** [comment]
Hi could you add an MRE? Public repo or sandbox we can run and reproduce the issue

**Reply by VictorBlockchain** [reply]
nothing works bro... can't connect to trust wallet, can't connect to metamask... lost enough time... got it working in 15 mins with metamask sdk...

**Comment by glitch-txs** [comment]
Close for lack of information, please provider a Minimal Reproducible Example and reopen the issue.
